Looks like it failed with
```
2024-04-23T10:37:26.1468304Z [INFO] 16479 goals, 14859 executed, 1620 from cache
2024-04-23T10:37:26.4654745Z [ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.2.1:enforce (enforce) on project quarkus-integration-test-webjars-locator: 
2024-04-23T10:37:26.4657031Z [ERROR] Rule 1: io.quarkus.enforcer.RequiresMinimalDeploymentDependency failed with message:
2024-04-23T10:37:26.4658833Z [ERROR] 1 minimal *-deployment dependencies are superfluous and must be removed from pom.xml:
2024-04-23T10:37:26.4660315Z [ERROR]     io.quarkus:quarkus-webjars-locator-deployment:999-SNAPSHOT
```

Let me pick up the latest commits, I thought I did

--- a/integration-tests/bouncycastle-fips/src/main/java/io/quarkus/it/bouncycastle/BouncyCastleFipsEndpoint.java
+++ b/integration-tests/bouncycastle-fips/src/main/java/io/quarkus/it/bouncycastle/BouncyCastleFipsEndpoint.java
@@ -1,12 +1,22 @@
 package io.quarkus.it.bouncycastle;
 
+import java.security.SecureRandom;
 import java.security.Security;
 import java.security.Signature;
 import java.util.Arrays;
 import java.util.stream.Collectors;
 
+import javax.crypto.KeyGenerator;
+
 import jakarta.ws.rs.GET;
 import jakarta.ws.rs.Path;
+
+import org.bouncycastle.crypto.CryptoServicesRegistrar;
+import org.bouncycastle.crypto.EntropySourceProvider;
+import org.bouncycastle.crypto.fips.FipsDRBG;
+import org.bouncycastle.crypto.fips.FipsUnapprovedOperationError;
+import org.bouncycastle.crypto.util.BasicEntropySourceProvider;
+import org.bouncycastle.util.Strings;
 
 @Path("/jca")
 public class BouncyCastleFipsEndpoint {
@@ -25,5 +35,33 @@ public class BouncyCastleFipsEndpoint {
         // This algorithm name is only supported with BC, Java (11+) equivalent is `RSASSA-PSS`
         Signature.getInstance("SHA256withRSAandMGF1", "BCFIPS");
         return "success";
+    }
+
+    @GET
+    @Path("fipsmode")
+    public String confirmFipsMode() throws Exception {
+        // https://www.bouncycastle.org/fips-java/BCFipsIn100.pdf
+
+        // Ensure that only approved algorithms and key sizes for FIPS-140-3.
+        CryptoServicesRegistrar.setApprovedOnlyMode(true);
+        // Set Secure Random to be compliant
+        EntropySourceProvider entSource = new BasicEntropySourceProvider(new SecureRandom(), true);
+        FipsDRBG.Builder drgbBldr = FipsDRBG.SHA512
+                .fromEntropySource(entSource)
+                .setSecurityStrength(256)
+                .setEntropyBitsRequired(256);
+        CryptoServicesRegistrar.setSecureRandom(drgbBldr.build(Strings.toByteArray("axs"), true));
+
+        // Validates FIPS Mode enabled and enforced correctly with Unapproved Key Generation
+        KeyGenerator keyGenerator = KeyGenerator.getInstance("HmacSHA512", "BCFIPS");
+
+        try {
+            keyGenerator.init(256);
+            return "HMAC SHA-512 initialization should not work when FIPS enabled.";
+        } catch (FipsUnapprovedOperationError ex) {
+            return "HMAC SHA-512 initialization does not work when FIPS enabled.";
+        } catch (Exception exception) {
+            return exception.getClass().getName();
+        }
     }
 }

--- a/integration-tests/bouncycastle-fips/src/test/java/io/quarkus/it/bouncycastle/BouncyCastleFipsTestCase.java
+++ b/integration-tests/bouncycastle-fips/src/test/java/io/quarkus/it/bouncycastle/BouncyCastleFipsTestCase.java
@@ -30,4 +30,13 @@ public class BouncyCastleFipsTestCase {
                 .body(equalTo("success"));
     }
 
+    @Test
+    public void testFipsMode() {
+        RestAssured.given()
+                .when()
+                .get("/jca/fipsmode")
+                .then()
+                .statusCode(200)
+                .body(equalTo("HMAC SHA-512 initialization does not work when FIPS enabled."));
+    }
 }

